src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -58,6 +60,64 @@ class PerformanceProblem:
     cause_span_ids: Optional[Sequence[str]]
     offender_span_ids: Sequence[str]
 
+    def serialize(self) -> str:
+        return json.dumps(
+            {
+                "fingerprint": self.fingerprint,
+                "op": self.op,
+                "desc": self.desc,
+                "type": self.type.value,
+                "parent_span_ids": self.parent_span_ids,
+                "cause_span_ids": self.cause_span_ids,
+                "offender_span_ids": self.offender_span_ids,
+            }
+        )
+
+    @classmethod
+    def deserialize(cls, serialized: str) -> PerformanceProblem:
+        data = json.loads(serialized)
+        return cls(
+            data["fingerprint"],
+            data["op"],
+            data["desc"],
+            GroupType(data["type"]),
+            data["parent_span_ids"],
+            data["cause_span_ids"],
+            data["offender_span_ids"],
+        )
+
+
+class EventPerformanceProblem:
+    """
+    Wrapper that binds an Event and PerformanceProblem together and allow the problem to be saved
+    to and fetch from Nodestore
+    """
+
+    def __init__(self, event: Event, problem: PerformanceProblem):
+        self.event = event
+        self.problem = problem
+
+    @property
+    def identifier(self) -> str:
+        return self.build_identifier(self.event.event_id, self.problem.fingerprint)
+
+    @classmethod
+    def build_identifier(cls, event_id: str, problem_hash: str) -> str:
+        identifier = hashlib.md5(f"{problem_hash}:{event_id}".encode()).hexdigest()
+        return f"p-i-e:{identifier}"
+
+    def save(self):
+        nodestore.set(self.identifier, self.problem.serialize())
+
+    @classmethod
+    def fetch(cls, event: Event, problem_hash: str) -> EventPerformanceProblem:
```

Comment:
We could potentially just require an `event_id` here and in the constructor. Might be more flexible?